### ISSUE TITLE
Copyright Notice Correct In All Files

### ISSUE THREAD

**Issue by davelab6** [googlefonts/fontbakery#183]
Write a Upstream test, to check that the Copyright Notices are each
- [x] identical
- [x] following the correct pattern, according a regex ~~of `Copyright * NNNN * (*@*.*)*`~~
- [x] match those in the binaries

**Comment by davelab6** [comment]
This test (and likely others) should be abstracted to work on each test pathway (UFO/TTF/etc)

**Comment by davelab6** [comment]
@m4rc1e does your 'upstream' GlyphsApp preflight script check this?

**Comment by davelab6** [comment]
@m4rc1e does your 'upstream' GlyphsApp preflight script check this?

**Comment by m4rc1e** [comment]
It will check if it meets the following structure:
`Copyright YYYY the {{ font.familyName }} Project Authors (git url)`

I still need it to check that the git url is valid and the master source :-)

**Comment by davelab6** [comment]
Should be

    Copyright YYYY The {{ font.familyName }} Project Authors ({{ git url }})

:)

**Comment by davelab6** [comment]
It occurs to me that the 3 things listed above specify a 'dashboard' milestone check, comparing source to live binaries, and therefore a check across the push life cycle. 

Therefore when you've fixed your source checker, I'll assign Felipe and move it to that milestone.

**Comment by davelab6** [comment]
@felipesanches do we have this in the TTF checker?

**Comment by m4rc1e** [comment]
@davelab6 I have this in the gf-glyph-scripts check, https://github.com/googlefonts/gf-glyphs-scripts/blob/master/Google%20Fonts/qa.py#L172-L204

**Comment by felipesanches** [comment]
thanks, @m4rc1e

**Comment by m4rc1e** [comment]
@davelab6 gf-glyph-scripts can now auto generate the copyright string for both the OFL.txt and font copyright, https://github.com/googlefonts/gf-glyphs-scripts/blob/master/Google%20Fonts/fixfonts.py#L30-L64.

The family needs to be included in the GF Upstream Repo doc in order for this to work. There is also a test to make sure the family is included in this document.

**Comment by davelab6** [comment]
@felipesanches do we have this in the TTF checker?

We'll need that for https://github.com/google/fonts/issues/1134 - simple grep can take care of finding emails in DESC and METADATA files.

**Comment by felipesanches** [comment]
We do have a test for this but with diverging set of rules and only applying to METADATA.pb.
This is **com.google.fonts/test/102** ("METADATA.pb: Copyright notice matches canonical pattern ?") and it will:

* **PASS** for fonts with the string "Copyright 2016 Author Name (name@site.com)"
* **WARN** that the copyright notice is OK, but lacks an email, when finding the string "Copyright 2016 Author Name"
* **FAIL** for any other pattern.

In the description of this issue, when you say that copyright notices should "match those in the binaries", where should they be (for instance, in which name table entries ?) We have a rule against copyright info into description strings in the name table at **com.google.fonts/test/031** ("Description strings in the name table must not contain copyright info."). And at **com.google.fonts/test/088** ("Copyright notice is the same in all fonts ?") we make sure all font.copyright fields are coherent within METADATA.pb.

Would you like me to change the regex patterns on **test/102** ?
And then, how to accomodate test/031 requirements that seem contradictory with the request in this issue? Should we drop (or change) its rule against copyright info on description strings of the name table?

**Comment by felipesanches** [comment]
Maybe the correct place for copyright strings on the name table would be on NAMEID_COPYRIGHT_NOTICE. I think this is possibly the correct place to look for that info.

If you acknowledge that, then I think we need to only do 2 things:
* create a new test for making sure NAMEID_COPYRIGHT_NOTICE name entries match font.copyright on METADATA.pb
* Change the regex on the **test/102** validation of METADATA.pb font.copyright entries.

**Comment by felipesanches** [comment]
I created a new test (**com.google.fonts/test/155**) to make sure NAMEID_COPYRIGHT_NOTICE is identical to METADATA.pb font.copyright entries

**Comment by davelab6** [comment]
Yes, the regex marc's gf-glyphs-script checker uses is preferred, please
update

**Comment by davelab6** [comment]
No, test 031 is still required, we don't want copyright notices in font
description fields.

**Comment by felipesanches** [comment]
![screenshot at 2017-08-11 19 23 16](https://user-images.githubusercontent.com/213676/29233992-8bd795d4-7eca-11e7-8e4d-d2b5de7ee401.png)

![screenshot at 2017-08-11 19 23 57](https://user-images.githubusercontent.com/213676/29234021-a877f86e-7eca-11e7-8277-31f58289d8ae.png)

![screenshot at 2017-08-11 19 30 35](https://user-images.githubusercontent.com/213676/29234177-95f54970-7ecb-11e7-8a98-5d43935d22ba.png)

**Comment by davelab6** [comment]
Mountains of C is a super fail because of the RFN...

**Comment by davelab6** [comment]
> create a new test for making sure NAMEID_COPYRIGHT_NOTICE name entries match font.copyright on METADATA.pb

That is back to font; font.copyright in METADATA is originally extracted from NAMEID_COPYRIGHT_NOTICE, so we should check that the child METADATA values are matching the parent NAMEID_COPYRIGHT_NOTICE values. 

Also test 031 is still required but your screenshot doesn't show that working

**Comment by felipesanches** [comment]
Reopening to think about what you just said.
